### PR TITLE
eager_loading association without primary key

### PR DESCRIPTION
**Pull request by chopraanmol1**
Fix for #29374

This will allow eager_loading models without any primary key

### CODE REVIEW

File: activerecord/lib/active_record/associations/join_dependency.rb [hunk 1]
```diff
@@ -241,15 +241,18 @@ def construct(ar_parent, parent, row, rs, seen, model_cache, aliases)
...
             id = row[key]
             if id.nil?
               nil_association = ar_parent.association(node.reflection.name)
               nil_association.loaded!
               next
             end
 
-            model = seen[ar_parent.object_id][node.base_klass][id]
+            # if a model does not have primary key than we are using foreign key
+            # which is not unique so we will avoid anytype of cache
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/associations/join_dependency.rb]: `anytype` -> `any type`

**Review by rafaelfranca — APPROVED** [review]
LGTM. r? @sgrif

**Comment by chopraanmol1** [comment]
Any updates on this PR

**Comment by eagletmt** [comment]
I also faced the same issue and confirmed this PR solves it.

Rails version: 5.1.4
Ruby version: 2.4.2
Database: PostgreSQL 9.6.5

**Pull request closed without merging**